stumpy/cache.py

```diff
@@ -107,3 +109,40 @@ def _get_cache():
     site_pkg_dir = site.getsitepackages()[0]
     numba_cache_dir = site_pkg_dir + "/stumpy/__pycache__"
     return [f.name for f in pathlib.Path(numba_cache_dir).glob("*nb*") if f.is_file()]
+
+
+def _recompile(func=None, fastmath=None):
+    """
+    Recompile a jit/njit decorated function. If `func` is None, then it wiil
+    recompile all njit functions of STUMPY.
+
+    Parameters
+    ----------
+    func : a njit function, default None
+        The numba function to recompile. If None, then all njit functions
+        of STUMPY will be recompiled.
+
+    fastmath : bool or set, default None
+        The fastmath flags to use. If None, then the func's fastmath flags
+        will not be changed. This is only used when `func` is provided.
+
+    Returns
+    -------
+    None
+    """
+    warnings.warn(CACHE_WARNING)
+    if func is None:
+        njit_funcs = get_njit_funcs()
+        for module_name, func_name in njit_funcs:
+            module = importlib.import_module(f".{module_name}", package="stumpy")
+            func = getattr(module, func_name)
+            func.recompile()
```

Comment:
Don't you need to handle the `fastmath` parameter here too?

```
if fastmath is not None:
    func.targetoptions["fastmath"] = fastmath
```

Reply:
Yes! I noticed you considered it in the code you proposed in another comment.

---

The reason I decided to go to this direction was that I did not want to allow users to use one set of fastmath flags on all njit functions. I thought this would force users to think what they are doing. And if they really want to do it, they can just write a for-loop and calls the function `cache._recompile` there. But that is too much hassle for them. So, when I read the previous two sentences again and again, I can see that allowing users to use one fastmath flag should be OK.